tests/integration/test_scaling.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import time
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+from scaling_helpers import check_key, get_password, restart_unit, srvr, write_key
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+async def ping_servers(ops_test: OpsTest):
+    logger.info("pinging servers")
+    for unit in ops_test.model.applications[APP_NAME].units:
+        host = unit.public_address
+        logger.info(f"{srvr(host)=}")
+        mode = srvr(host)["Mode"]
+        assert mode in ["leader", "follower"]
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_active(ops_test: OpsTest):
+    charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(charm, application_name=APP_NAME, num_units=3)
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[APP_NAME].units) == 3)
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
```

Comment:
why set the update status hook and reset it? I've seen this before but I can't remember where/why

Reply:
I think it's done so it triggers quickly within this test, but afterwards you don't have update_status triggering all the time.

Reply:
You can actually replace this with a nice short context manager now :) 

https://github.com/jnsgruk/kubernetes-dashboard-operator/blob/dc580a9a205e5369a319fd55ecdf83d25bc66450/tests/integration/test_charm.py#L45